<?xml version="1.0"?>
<!--for usv, get yaw from imu-->
<launch>
     <!--guidance  /-->
	<node name="map_guidance_node" pkg="g_usv" type="map_g_guidance_node"  output="screen">
		<remap from="/gnc/c_heading" to="/control/heading" />
	</node>
 <node name="heartbeat_sender" pkg="mz_usv" type="heartbeat_sender" />
     <!--odometry  /-->
		<!--usv  /-->
	<node name="auto_pwm" pkg="mz_usv" type="auto_pwm" launch-prefix="sudo -E"> 
	</node>
	<node name="battery_sensor" pkg="mz_usv" output="screen"  type="battery_sensor"  args="2 4800" /> 
	<node name="imu" pkg="mz_usv" type="imu" >
	</node>
	<node name="imu_filter_node" pkg="imu_filter_madgwick" type="imu_filter_node" >
		<param name = "constant_dt"  value ="0.0"/>
		<remap from="imu/data" to="/sensor/m_imu"/>
		<param name = "world_frame"  value ="enu"/>	
		<param name = "use_mag"  value ="true"/>	
		<param name = "use_magnetic_field_msg"  value ="true"/>		
	</node>
	<node name="heading_PID" pkg="mz_usv"  type="heading_PID" >
		<param name = "K_p"  value ="40.0"/>
		<param name = "K_i"   value ="0.01"/>
		<param name = "K_d"   value ="20.0"/>
		<param name = "K_g"   value ="0.001"/>
		<param name = "Total_thrust" value ="1.6"/>
		<remap from="/imu/data" to="/sensor/m_imu" />
	</node>
	<!--gps  /-->
  <arg name="port" default="/dev/ttyS2" />
  <arg name="baud" default="115200" />
  <arg name="frame_id" default="gps" />
  <arg name="use_GNSS_time" default="False" />
  <arg name="time_ref_source" default="gps" />
  <arg name="useRMC" default="True" />

  <node name="nmea_serial_driver" pkg="nmea_navsat_driver" type="nmea_serial_driver">
    <param name="port" value="$(arg port)"/>
    <param name="baud" value="$(arg baud)" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="use_GNSS_time" value="$(arg use_GNSS_time)" />
    <param name="time_ref_source" value="$(arg time_ref_source)" />
	<remap from="/fix" to="/sensor/m_fix" />
	<remap from="/vel" to="/sensor/m_vel" />
    <param name="useRMC" value="$(arg useRMC)" />
  </node>
  	
  	<!--node pkg="master_discovery_fkie" type="master_discovery" name="master_discovery">
		<param name="mcast_group" value="224.0.0.1"/>
	</node>
	<node name="my_master_sync_fkie" pkg="master_sync_fkie" type="master_sync">
	<rosparam param="sync_topics">['/joy']</rosparam>
	<rosparam param="sync_topics">['/mapviz/wayPoints']</rosparam>
	</node-->
 <node pkg="mz_usv" type="tcp_server" name="tcp_server" args="8000" />
</launch>
